syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.routeguide";
option java_outer_classname = "RouteGuideProto";
option objc_class_prefix = "RTG";

package routeguide;

// Interface exported by the server.
service RouteGuide {
  rpc UploadImage(stream ImageChunk) returns (UploadStatus) {}
  rpc BidirectionalImageTransfer(stream ImageChunk) returns (stream ImageChunk) {}
}

message ImageChunk {
  bytes data = 1;
}

message UploadStatus {
  bool success = 1;
  string message = 2;
}
